test/cfg/googletest.cpp

```diff
@@ -47,6 +47,32 @@ TEST(test_cppcheck, cppcheck)
 // #9964 - avoid compareBoolExpressionWithInt false positive
 TEST(Test, assert_false_fp)
 {
-    // cppcheck-suppress checkLibraryNoReturn
     ASSERT_FALSE(errno < 0);
 }
+
+// Check that conditions in the ASSERT_* macros are processed correctly.
+TEST(Test, warning_in_assert_macros)
+{
+    int i = 5;
+    int j = 6;
+
+    // cppcheck-suppress knownConditionTrueFalse
+    ASSERT_TRUE(i == 5);
+    // cppcheck-suppress knownConditionTrueFalse
+    ASSERT_FALSE(i != 5);
+    // cppcheck-suppress duplicateExpression
+    ASSERT_EQ(i, i);
+    ASSERT_NE(i, j); // expected knownConditionTrueFalse...
```

Comment:
There seems to be a bug with knownConditionTrueFalse and macros. It works if I replace the macro with the exact text from the cfg file, so I'm guessing it's unrelated to my changes (also didn't work with the asserts).